src/memory-usage.ts

```diff
@@ -0,0 +1,109 @@
+// Implements the "memory usage" feature.
+// When the server advertises `memoryUsageProvider`, a command
+// (clangd.memoryUsage) is available (context variable:
+// clangd.memoryUsage.supported). It sends the $/memoryUsage request and
+// displays the result in a tree view (clangd.memoryUsage) which becomes visible
+// (context: clangd.memoryUsage.hasData)
+
+import * as vscode from 'vscode';
+import { window } from 'vscode';
+import * as vscodelc from 'vscode-languageclient/node';
+
+import {ClangdContext} from './clangd-context';
+
+export function activate(context: ClangdContext) {
+  const feature = new ASTFeature(context);
+  context.client.registerFeature(feature);
+}
+
+// LSP wire format for this clangd feature.
+interface NoParams {}
+interface WireTree {
+  _self: number;
+  _total: number;
+  [child: string]: WireTree|number|string;
```

Comment:
my typescript isnt so strong, but why can this be anything but WireTree ? isn't it enough to handle `_self` and `_total` explicitly? (and even if that's not enough, where does the string come from)

Reply:
_self and _total have to conform to this constraint too, unfortunately. I couldn't find a way to more precisely express the type. This is one of the reasons for mapping quickly to another type :-)

(There's a workaround involving intersection types, but they must be `type` rather than `interface` and `type`s can't be recursive as they're just aliases)